x-pack/platform/plugins/shared/observability_ai_assistant/server/functions/index.ts

```diff
@@ -44,102 +43,43 @@ export const registerFunctions: RegistrationCallback = async ({
-        instructions.push(`You can use the "${SUMMARIZE_FUNCTION_NAME}" function to store new information you have learned in a knowledge database.
-          If the user asks to remember or store some information, always use this function.
-          All summaries MUST be created in English, even if the conversation was carried out in a different language.`);
-      }
-
-      if (availableFunctionNames.includes(CONTEXT_FUNCTION_NAME)) {
-        instructions.push(
-          `You can use the "${CONTEXT_FUNCTION_NAME}" function to retrieve relevant information from the knowledge database. The response will include a "learnings" field containing information
-          from the knowledge base that is most relevant to the user's current query. You should incorporate these learnings into your responses when answering the user's questions.
-          The information in the "learnings" field contains up-to-date information that you should consider when formulating your responses. DO NOT add disclaimers about the currency or certainty of this information.
-          Present this information directly without qualifiers like "I don't have specific, up-to-date information" or "I can't be completely certain".
-          
-          Stick strictly to the information provided in the "learnings" field. DO NOT assume, infer, or add any details that are not explicitly stated in the response.
-          If the user asks for information that is not covered in the "learnings" field, acknowledge the gap and ask for clarification rather than making assumptions or offering suggestions that aren't based on the provided knowledge.`
-        );
-      }
-    } else {
-      instructions.push(
-        `You do not have a working memory. If the user expects you to remember the previous conversations, tell them they can set up the knowledge base.`
-      );
-    }
-    return instructions.map((instruction) => dedent(instruction));
-  });
+  // determine if product documentation is installed
+  const llmTasks = pluginsStart?.llmTasks;
+  const esClient = (await resources.context.core).elasticsearch.client;
+  const inferenceId =
+    (await getInferenceIdFromWriteIndex(esClient, resources.logger)) ??
+    defaultInferenceEndpoints.ELSER;
+  const isProductDocAvailable = llmTasks
+    ? await llmTasks.retrieveDocumentationAvailable({ inferenceId })
+    : false;
+
+  functions.registerInstruction(({ availableFunctionNames }) =>
+    getSystemPrompt({
+      availableFunctionNames,
+      isServerless,
+      isKnowledgeBaseReady,
+      isObservabilityDeployment,
+      isGenericDeployment,
+      isProductDocAvailable,
+    })
+  );
 
   if (isKnowledgeBaseReady) {
     registerSummarizationFunction(registrationParameters);
+  } else {
+    functions.registerInstruction(
+      `You do not have a working memory. If the user expects you to remember the previous conversations, tell them they can set up the knowledge base.`
```

Comment:
This make it sounds like we query past conversations but we only (currently) query the knowledge base.

```suggestion
      `You do not have a working memory. If the user expects you to use retrieve documents from the knowledge base, tell them they can set up it up first.`
```